### PR TITLE
fix: update browser examples to use async check

### PR DESCRIPTION
**Pull request by VikaCep**
Updates browser checks examples to use `async check` from `https://jslib.k6.io/k6-utils/1.5.0/index.js` as recommended in the [FAQ document](https://docs.google.com/document/d/1SbBdhYVGnEQ9XJe6DKYMErPI3O_E3anAp3Hdu_faUo8/edit#heading=h.duuadjcekgl6)

image

image

### CODE REVIEW

File: src/components/BrowserExamplesMenu/snippets/dispatch.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default async function() {
 
     const h3 = page.locator("h3");
     const ok = await h3.textContent() == "Contact us";
-    check(ok, { "header": ok });
+    await check(ok, { "header": ok });
```
Comment (ckbedwell) [inline src/components/BrowserExamplesMenu/snippets/dispatch.js:32]: I think we should change some of these snippets to match the intent of the async/await so it is clearer what it is doing:

```
    await check(page.locator("h3"), {
      header: async (lo) => (await lo.textContent()) === "Contact us",
    })
```

File: src/components/BrowserExamplesMenu/snippets/getattribute.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default async function () {
     });
     let el = await page.$('#dark-mode-toggle-3');
     const mode = await el.getAttribute('mode');
-    check(mode, {
+    await check(mode, {
       "GetAttribute('mode')": mode === 'light',
     });
```
Comment (ckbedwell) [inline src/components/BrowserExamplesMenu/snippets/getattribute.js:32]:
```
await check(page.locator("#dark-mode-toggle-3"), {
      "GetAttribute('mode')": async (lo) =>
        (await lo.getAttribute("mode")) === "light",
    })
    ```

File: src/components/BrowserExamplesMenu/snippets/querying.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ export default async function () {
     const titleWithCSS = await page.$('header h1.title').then((e) => e.textContent());
     const titleWithXPath = await page.$(`//header//h1[@class="title"]`).then((e) => e.textContent());
```
Comment (ckbedwell) [inline src/components/BrowserExamplesMenu/snippets/querying.js:28]: Same here (and get rid of the `page.$` and replace with `page.locator`)

File: src/components/BrowserExamplesMenu/snippets/waitForFunction.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ export default async function () {
       polling: 'mutation',
       timeout: 2000,
     });
-    check(ok, { 'waitForFunction successfully resolved': ok.innerHTML() == 'Hello' });
+    await check(ok, { 'waitForFunction successfully resolved': ok.innerHTML() == 'Hello' });
   } finally {
     await page.close();
   }
```
Comment (ckbedwell) [inline src/components/BrowserExamplesMenu/snippets/waitForFunction.js:40]: Total side note but I hate [this waitForFunction example](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/page/waitforfunction/) 😂 @ankur22 @inancgumus do we have a better example than this? It feels like it is giving mixed messages how to do page locators.

I like the [Playwright equivalent example](https://playwright.dev/docs/api/class-page#page-wait-for-function).

**Review by ckbedwell — CHANGES_REQUESTED** [review]
Just some thoughts! What do you think?

**Comment by VikaCep** [comment]
Good points @ckbedwell, I changed the examples you mentioned as well as the the script we display when creating a new check 👍

**Review by ckbedwell — APPROVED** [review]
LGTM 🚀

**Pull request merged**